### PR TITLE
Benchmark user session hash function - to help with string casting ticket

### PR DESCRIPTION
**Pull request by joshblakeley**
As suggested by @dencoded in comments of #1680 

I have done this as it's own PR so it can be merged to master separately as it is expansion of benchmarking rather than part of a fix.

### CODE REVIEW

File: user/session_test.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package user
+
+import "testing"
+
+var session SessionState
+
+func BenchmarkHash(b *testing.B) {
```
Comment (dencoded) [inline user/session_test.go]: need to enable memory allocations with `b.ReportAllocs()`

File: user/session_test.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package user
+
+import "testing"
+
+var session SessionState
```
Comment (dencoded) [inline user/session_test.go]: might worth to create session with some data instead of zero-value struct

File: user/session_test.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+func BenchmarkHash(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		session.Hash()
```
Comment (dencoded) [inline user/session_test.go]: need to save result into variable out of scope to avoid compiler optimizations

**Comment by joshblakeley** [comment]
Thanks @dencoded will improve tomorrow

**Pull request merged**